### PR TITLE
HRINT-2429 [Demo] Static Indexes: Fanout Map Index

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

**Comment by Danielle9897** [comment]
@DejanMilicic can you pls review ?

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
+{
+    "Slug": "fanout-index",
+    "SourceFileName": "FanoutIndexController.cs",
+    "Title": "Fanout Index",
+    "DescriptionLines": [
+        "When a document field is an array, the inner array fields can be indexed in what is called a **Fanout Index.**",
+        "",
+        "An index entry will be generated per item of the array,",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:8]: And after that, we can explain that one of such cases is processing of array properties, where each element of an array will result in one index entries

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            "Title": "The Index Map",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* The _Map_ function defines the index-fields content - which fields from the 'Order' document will compose each index-field.",
+                "",
+                "* In this example, an order document has a _Lines_ array object:<br>",
+                "  To access the inner fields of the Lines array use: `from orderLine in order.Lines`.",
+                "",
+                "* An index entry with fields _ProductId_ & _ProductName_ will be created per item of the Lines array,<br>",
+                "  so the number of entries created per Order document is as the array length.",
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:102]: Maybe change to "so the number of entries created per Order document equals the array length"?

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        },
+        {
+            "Title": "Query for documents",
+            "Slug": "step-4",
+            "DescriptionLines": [
+                "* Query the 'Orders' collection using the defined index.",
+                "",
+                "* RavenDB will search the index for terms that match the query criteria and retrieve all matching documents.",
+                "",
+                "* The query results will include all documents from the 'Orders' collection that have an array field 'ProductName' that starts with 'namePrefix'."
```
Comment (DejanMilicic) [inline DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:133]: "array field" -> "Lines array field" ?

**Review by DejanMilicic — CHANGES_REQUESTED** [review]

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json (hunk 2 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:102]: done

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:133]: done

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json [hunk 4]
```diff
@@ -0,0 +1,159 @@
+{
+    "Slug": "fanout-index",
+    "SourceFileName": "FanoutIndexController.cs",
+    "Title": "Fanout Index",
+    "DescriptionLines": [
+        "When a document field is an array, the inner array fields can be indexed in what is called a **Fanout Index.**",
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:6]: image

File: DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json (hunk 1 above)
Comment (Danielle9897) [reply DemoServer/Controllers/Demos/StaticIndexes/FanoutIndex/metadata.json:8]: see above comment

**Review by DejanMilicic — APPROVED** [review]

**Comment by Danielle9897** [comment]
Resolved conflicts - Can be merged
@topalach @DejanMilicic

**Review by DejanMilicic — APPROVED** [review]

**Pull request merged**